pkg/plugin/plugin.go

```diff
@@ -95,7 +97,16 @@ func ParseError(err error) (errorType KMSErrorType) {
 	// ref. https://docs.aws.amazon.com/kms/latest/developerguide/requests-per-second.html
 	case kms.ErrCodeLimitExceededException:
 		return KMSErrorTypeThrottled
+
+	// e.g., "AccessDeniedException: The ciphertext refers to a customer master key that does not exist, does not exist in this region, or you are not allowed to access."
+	case "AccessDeniedException":
+		if strings.Contains(ev.Message(), "customer master key that does not exist") &&
```

Comment:
I don't feel super comfortable relying on specific message strings in an error. Are there other documented error strings for this error type?

Reply:
Let me consult with KMS team :)

Reply:
FWIW, we discovered this error message generated by decrypt ciphertext with a different key ID from the one encrypt the secret. From the KMS public doc, looks like explicitly set key ID in decrypt API input is a best practice. 

> If the ciphertext was encrypted under a symmetric KMS key, the KeyId parameter is optional. AWS KMS can get this information from metadata that it adds to the symmetric ciphertext blob. This feature adds durability to your implementation by ensuring that authorized users can decrypt ciphertext decades after it was encrypted, even if they've lost track of the key ID. However, specifying the KMS key is always recommended as a best practice. When you use the KeyId parameter to specify a KMS key, AWS KMS only uses the KMS key you specify. If the ciphertext was encrypted under a different KMS key, the Decrypt operation fails. This practice ensures that you use the KMS key that you intend. 

ref. https://docs.aws.amazon.com/kms/latest/APIReference/API_Decrypt.html

I believe in that case, KMS API will return a corresponding ErrCode.

Reply:
Confirmed with KMS team. The error code won't change but the message may change. So, we need string match. And the current SDK Go does not define any specific error code or message for this error.